src/creatures/combat/condition.cpp

```diff
@@ -1460,6 +1496,276 @@ void ConditionDamage::generateDamageList(int32_t amount, int32_t start, std::lis
 	}
 }
 
+/**
+ *  ConditionFeared
+ */
+bool ConditionFeared::isStuck(Creature* creature, Position pos) {
+	for (Direction dir : dirList) {
+		if (canWalkTo(creature, pos, dir)) {
+			return true;
+		}
+	}
+
+	return false;
+}
+
+bool ConditionFeared::getRandomDirection(Creature* creature, Position pos) {
+	std::shuffle(dirList.begin(), dirList.end(), getRandomGenerator());
```

Comment:
shuffling the shared vector of the object will impact in other functions that uses `fleeIndx`

Reply:
Maybe using the same strategy as in ```Monster::getRandomStep```  may solve this